x/stakeibc/keeper/icqcallbacks_validator_exchange_rate.go

```diff
@@ -115,17 +115,22 @@ func ValidatorExchangeRateCallback(k Keeper, ctx sdk.Context, args []byte, query
 		k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Validator,
 			"Validator was not slashed, finishing LSM liquid stake"))
 
-		async := true
-		if err := k.FinishLSMLiquidStake(ctx, lsmLiquidStake, async); err != nil {
-			// If finishing the transaction failed, emit an event, but swallow the error so
-			// the query object still gets deleted
+		// Finish the LSMLiquidStake with a temporary context so that the state changes can
+		// be discarded if it errors
+		err := utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
+			async := true
```

Comment:
this is the only place we need it. the only reason I stored it as a variable was for readability purposes since there aren't named params in golang and simply passing `true` would be cryptic